os/customizing-docker.md

```diff
@@ -66,6 +66,123 @@ To keep access to the port local, replace the `ListenStream` configuration above
+
+```sh
+# Initialize Certificate Authority
+./etcd-ca init --passphrase=""
+# Create certificate request for %hostname% docker server
+./etcd-ca new-cert --passphrase="" --domain="example.com" --ip="%private_ip%,%public_ip%" %hostname%
+# Sign certificate request for %hostname% docker server
+./etcd-ca sign --passphrase="" %hostname%
+# Export docker server certificate and key without password protection
+./etcd-ca export --passphrase="" --insecure %hostname% | tar xvf -
+# Create certificate request for docker client
+./etcd-ca new-cert --passphrase="" client
+# Export client certificate and key without password protection
+./etcd-ca export --passphrase="" --insecure client | tar xvf -
+```
+
+As a result you'll get following files:
+
+```sh
+%hostname%.crt
+%hostname%.key.insecure
+client.crt
+client.key.insecure
+.etcd-ca/ca.crt
+```
+
+Copy following files into `/etc/docker` CoreOS' directory and fix their permissions:
+
+```sh
+sudo cp %hostname%.crt %hostname%.key.insecure .etcd-ca/ca.crt /etc/docker/
+sudo chmod 0600 %hostname%.key.insecure
+sudo chmod 0644 %hostname%.crt ca.crt
+```
+
+Copy certificates and key for client into your local machine and rename them:
+
+```
+mkdir ~/.docker
+cd ~/.docker
+scp core@%hostname%:{client.crt,client.key.insecure,.etcd-ca/ca.crt} .
+mv ca.crt ca.pem
+mv client.crt cert.pem
+mv client.key.insecure key.pem
+```
+
+### Drop-in Configuration
+
+Create `/etc/systemd/system/docker.service.d/10-tls-verify.conf` [drop-in](using-systemd-drop-in-units.html) for systemd Docker service:
+
+```
+[Service]
+Environment="DOCKER_OPTS=-H=0.0.0.0:2376 --tlsverify --tlscacert=/etc/docker/ca.crt --tlscert=/etc/docker/%hostname%.crt --tlskey=/etc/docker/%hostname%.key.insecure"
+```
+
+Reload systemd config files and restart docker service:
+
+```sh
+systemctl daemon-reload
+systemctl restart docker
+```
+
+Now you can access your Docker's API through TLS secured connection:
+
+```sh
+docker --tlsverify -H tcp://%hostname%:2376 images
+# or
+docker --tlsverify -H tcp://example.com:2376 images
+```
+
+In addition you can export environment variables and use docker client without additional options:
+
+```sh
+export DOCKER_HOST=tcp://example.com:2376 DOCKER_TLS_VERIFY=1
+docker images
+```
+
+### Cloud-Config
+
+Cloud-config for Docker TLS authintication will look like:
+
+```yaml
+#cloud-config
+
+write_files:
+    - path: /etc/docker/ca.crt
+      permissions: 0644
+      content: |
+        ca.crt_content
+    - path: /etc/docker/ca.crt
+      permissions: 0644
+      content: |
+        %hostname%.crt_content
+    - path: /etc/docker/%hostname%.key.insecure
+      permissions: 0600
+      content: |
+        %hostname%.key.insecure_content
```

Comment:
Including a full key and cert would be too much text but perhaps you could truncate one to a couple lines to make it clear what is supposed to be there. people tend to copy examples verbatim so making it more obvious that they do indeed need to edit the example would be good.